Feature/improve playground layout

bulish opened the pull request.

ivopisarovic commented on vite.config.js:
```diff
@@ -0,0 +1,25 @@
+
+export default defineConfig({
+    plugins: [vue()],
+    root: "./src/vue/playground",
```
Why root is only the playground?

ivopisarovic commented on src/vue/playground/stores/PlaygroundStore.ts:
```diff
@@ -0,0 +1,96 @@
+    },
+  },
+  actions: {
+    async closeFile(
```
Quite messy code in this function, many nested conditions.

ivopisarovic commented:
@bulish I have reviewed your PR. Your code is in high quality and is very robust! :) Sadly, I have found several problems that block me from deploying this to production for now. 

1. When I create a new based on Vue, Vue is installed but there is still the `main.js` with the classic vanillajs code and also the preview does not show the Vue project but the classic vanilla example.
2. You are using two dependencies for syntax highlighting - `prism` and `shiki`. Can you please explain why two dependencies are needed? Prism has the last release already 2 years old...
3. Autocomplete apparently does not work for HTML in Javascript string literal. I am thinking about defining scene right in the `index.html` file in the Vanilla project and do only imports in `main.js`. This could make the thing easier for beginners. What do you think?
4. Does autocomplete already support A-FRAME and SDS elements? 
5. Installing dependencies takes a lot of time, maybe more than a minute. Is there any way to make it faster? For example, is it possible to pre-install node_modules and then only copy them to new containers?
6. There is an error on page http://localhost:5174/ar-vr-components/flexbox.html. Not sure if it is caused by your changes, but probably you inserted some new characters there by mistake. Please, fix it and check also other pages in docs.
7. There are quite many duplicate tailwind classes. No need to refactor it now but you should use more components for basic page elements in the future.
8. Also, you should not put so many changes as one pull request. Introducing ESLint and relevant changes should be a separate pull request. Otherwise, it is very hard to do code review. Keep it for now in one pull request, but do not forget to create smaller pull requests in the future.
9. Also, there are some comments directly in code, see above. 

Thanks for your work! :)

bulish commented:
@ivopisarovic 

Thank you for the review. I've fixed the issues you mentioned and pushed the changes to Git.

The changes are as follows:

1. Fixed: The vanilla project now contains A-Frame code in the main.js file, and the Vue.js project includes it in the App.vue file. The preview issue has also been resolved.
2. I have removed the Prism dependency and now exclusively use Shiki. That was a mistake I hadn’t noticed earlier.
3. Autocomplete for HTML in JavaScript string literals is now supported. There’s no problem creating the scene in the index.html file; I’ve simply followed the steps mentioned in the SDS documentation for creating new projects. If you’d prefer, I can ensure it’s done this way.
4. I have fixed the autocomplete for SDS and A-Frame. I decided to use a static list of components and attributes instead of dynamically fetching them from node_modules as before.
5. I’ve made some changes so that the playground now uses templates for both types of projects. After that, only npm install and npm run dev are executed. In the future, I plan to improve the loading experience between project creation and file display.
6. My apologies. To be honest, I’m not sure how it happened, but the error was caused by my changes. It has been fixed. The issues were in the Flexbox and Fit-Into-Fov components.
7., 8. Thank you for the feedback; I’ll keep that in mind.
9. Fixed.

ivopisarovic commented:
@bulish Thanks for fixing all the problems and for your detailed response! :) 

Just a quick recap:

1 ok
2 ok
3 great! autocomplete in literals works, so we can keep it this way
4 great!
5 It is much much better now! I think we can keep it, no need for further improvements. 
6 ok
7, 8, 9 ok

Sadly, I have some other comments, but only 2 remaining! :) 

- Where did you get the list of aframe primitives, attributes etc. for autocompletion? The file is missing a source link.
- I cannot edit any file. I change something in the editor but the preview still displays the original version. Also, when I reopen the file, there is the old code. Should I save the file somehow?

ivopisarovic commented:
@bulish told me that editing was fixed in another PR. 

Closing this now.
